text/0000-compiler-major-change-process.md

```diff
@@ -0,0 +1,255 @@
+* renaming a widely used, core abstraction, such as the `Ty` type
+* introducing cargo pipelining 
+* adding a new `-C` flag that exposes some minor variant
+
+## What are some examples of things that are too big for the major change process?
+
+Here are some examples of changes that are too big for the major change process, or which at least would require auxiliary design meetings or a more fleshed out design before they can proceed:
+
+* introducing incremental or the query system
+* introducing MIR or some new IR
+* introducing parallel execution
+* adding ThinLTO support
+
+## What are some examples of things that are too small for the major change process?
+
+Here are some examples of things that don't merit any MCP:
+
+* adding new information into metadata
+* fixing an ICE or tweaking diagnostics
+* renaming "less widely used" methods
+
+## When should Major Change Proposals be closed?
+
+Major Change Proposals can be closed:
+
+* by the author, if they have lost interest in pursuing it.
+* by a team lead or expert, if there are strong objections from key
+  members of the team that don't look likely to be overcome.
+* by folks doing triage, if there have been three months of
+  inactivity. In this case, people should feel free to re-open the
+  issue if they would like to "rejuvenate" it.
+
+# Template for major change proposals
+
+[major change template]: #Template-for-major-change-proposals
+
+The template for major change proposals is as follows:
+
+```
+# What is this issue?
+
+This is a **major change proposal**, which means a proposal to make a notable change to the compiler -- one that either alters the architecture of some component, affects a lot of people, or makes a small but noticeable public change (e.g., adding a compiler flag). You can read more about the MCP process on https://forge.rust-lang.org/.
+
+**This issue is not meant to be used for technical discussion. There is a Zulip stream for that. Use this issue to leave procedural comments, such as volunteering to review, indicating that you second the proposal (or third, etc), or raising a concern that you would like to be addressed.**
+
+# MCP Checklist
+
+* [x] Fill out and file this issue. The @rust-lang/wg-prioritization group will add this to the triage meeting agenda so folks see it.
+* [ ] Create a Zulip topic in the stream `#t-compiler/major changes` with the name `XXX compiler-team#NNN`, where `XXX` is the title of this issue and `NNN` is whatever issue number gets assigned. Discuss the MCP in there, and feel free to update the proposal as needed.
+* [ ] Find a **reviewer**, and add their name to this comment (see the section below).
+* [ ] Find a **second**, someone who is knowledgeable of the area and approves of the design, and have them leave a comment on the issue.
+* [ ] Announce this proposal at the triage meeting (ping @rust-lang/wg-prioritization to have them add it to the agenda). **All edits should be done before you do this.**
```

Comment:
Just going through this now (https://github.com/rust-lang/compiler-team/issues/278) and these instructions are a little unclear.

Does the word "ping" here mean to simply "mention" (include the text "@rust-lang/wg-prioritization") in a comment? 

I just did that, and this group/team did not get highlighted as I would have expected. Perhaps this group is not visible publicly?

If the submitter needs to be a member of an organization to see it, what org, and how do I join?

Thanks!

Reply:
I believe this is now automated via the "to-announce" label.

Reply:
Yes, the first step (creating a Zulip topic), and the 4th step (announcing the proposal) are automatically handled by automation or the team.

Reply:
Thank you. I was surprised to see these steps taken care of automatically. Can the template be updated for the two checkboxes (1st and 5th, actually, if I'm not mistaken) to say that the author is not responsible for these steps, or to say who is responsible (Rustbot or compiler team member)?

Thanks!